packages/salesforcedx-vscode-core/test/vscode-integration/orgBrowser/metadataOutlineProvider.test.ts

```diff
@@ -176,6 +176,46 @@ describe('load org browser tree outline', () => {
     getCmpsStub.restore();
   });
 
+  it('should load folder nodes when a folder containing metadata type node is selected', async () => {
+    const folderTypeMetadata = ['CustomObject', 'Dashboard', 'Document', 'EmailTemplate', 'Report'];
+    const expected = [
+      {
+        label: 'Dashboard1',
+        fullName: 'Dashboard1',
+        type: NodeType.Folder
+      },
+      {
+        label: 'Dashboard2',
+        fullName: 'Dashboard2',
+        type: NodeType.Folder
+      }
+    ];
+    const getCmpsStub = stub(
+      MetadataOutlineProvider.prototype,
+      'getComponents'
+      ).returns(expected.map(n => n.fullName));
+
+    const metadataObject = {
+      xmlName: 'foldertypeNode1',
+      directoryName: 'testDirectory',
+      suffix: 'cls',
+      inFolder: false,
+      metaFile: false,
+      label: 'Folder Type Node 1'
+    };
+    for (const folderMetadataType of folderTypeMetadata){
```

Comment:
It's better to remove for loops and "if" conditions in tests. Can we expand this test into multiple tests to test different folder metadata types?